src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -2875,184 +3102,1418 @@ private void ensurePublishCapability(
+                long traceId,
+                long authorization)
+            {
+                correlations.remove(replyId);
+
+                if (!MqttState.replyClosed(state))
+                {
+                    doApplicationReset(traceId, authorization);
+                }
+            }
+
+            private void setReplyClosed()
+            {
+                assert !MqttState.replyClosed(state);
+
+                state = MqttState.closeReply(state);
+
+                if (MqttState.closed(state))
+                {
+                    capabilities = 0;
+                    streams.remove(topicKey);
+                    final MutableInteger count = activeStreamsByTopic.get(topicKey);
+
+                    assert count != null;
+
+                    count.value--;
+
+                    assert count.value >= 0;
+
+                    if (count.value == 0)
+                    {
+                        activeStreamsByTopic.remove(topicKey);
+                    }
+                }
+            }
+
+            private void doApplicationFlush(
+                long traceId,
+                long authorization,
+                int reserved,
+                int flags)
+            {
+                initialBudget -= reserved;
+
+                assert initialBudget >= 0;
+
+                doFlush(application, routeId, initialId, traceId, authorization, 0L, reserved,
+                    ex -> ex.set((b, o, l) -> mqttFlushExRW.wrap(b, o, l)
+                                                           .typeId(mqttTypeId)
+                                                           .flags(flags)
+                                                           .capabilities(c -> c.set(valueOf(capabilities)))
+                                                           .build()
+                                                           .sizeof()));
+            }
+
+            private void setInitialClosed()
+            {
+                assert !MqttState.initialClosed(state);
+
+                state = MqttState.closeInitial(state);
+
+                if (debitorIndex != NO_DEBITOR_INDEX)
+                {
+                    debitor.release(debitorIndex, initialId);
+                    debitorIndex = NO_DEBITOR_INDEX;
+                }
+
+                if (MqttState.closed(state))
+                {
+                    capabilities = 0;
+                    streams.remove(topicKey);
+
+                    final MutableInteger activeStreams = activeStreamsByTopic.get(topicKey);
+
+                    assert activeStreams != null;
+
+                    activeStreams.value--;
+
+                    assert activeStreams.value >= 0;
+
+                    if (activeStreams.value == 0)
+                    {
+                        activeStreamsByTopic.remove(topicKey);
+                    }
+                }
+            }
+        }
+
+        private class MqttWillStream
```

Comment:
Since `will` stream is `publish-only` please remove the methods that are unused, such as on-app-data for the reply.